smash/core/simulation/optimize/optimize.py

```diff
@@ -691,6 +714,11 @@ def _adaptive_optimize(
                     )
                 break
 
+        if callback is not None:
```

Comment:
est ce que juste un `if callback` n'est pas suffisant ? Apres en soit ca peut etre plus clair avec `is not None`